operator/test/component/evaluationcontroller_test.go

```diff
@@ -0,0 +1,304 @@
+package component
+
+import (
+	"context"
+	"os"
+	"time"
+
+	klcv1alpha2 "github.com/keptn/lifecycle-toolkit/operator/apis/lifecycle/v1alpha2"
+	apicommon "github.com/keptn/lifecycle-toolkit/operator/apis/lifecycle/v1alpha2/common"
+	metricsv1alpha1 "github.com/keptn/lifecycle-toolkit/operator/apis/metrics/v1alpha1"
+	"github.com/keptn/lifecycle-toolkit/operator/controllers/common/providers"
+	"github.com/keptn/lifecycle-toolkit/operator/controllers/lifecycle/interfaces"
+	"github.com/keptn/lifecycle-toolkit/operator/controllers/lifecycle/keptnevaluation"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	otelsdk "go.opentelemetry.io/otel/sdk/trace"
+	sdktest "go.opentelemetry.io/otel/sdk/trace/tracetest"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+var _ = Describe("KeptnEvaluationController", Ordered, func() {
+	var (
+		evaluationName           string
+		evaluationDefinitionName string
+		metricName               string
+		namespaceName            string
+		spanRecorder             *sdktest.SpanRecorder
+		tracer                   *otelsdk.TracerProvider
+		ns                       *v1.Namespace
+	)
+
+	BeforeAll(func() {
+		// setup once
+		By("Waiting for Manager")
+		Eventually(func() bool {
+			return k8sManager != nil
+		}).Should(Equal(true))
+
+		By("Creating the Controller")
+		_ = os.Setenv("FUNCTION_RUNNER_IMAGE", "my-image")
+
+		spanRecorder = sdktest.NewSpanRecorder()
+		tracer = otelsdk.NewTracerProvider(otelsdk.WithSpanProcessor(spanRecorder))
+
+		////setup controllers here
+		controllers := []interfaces.Controller{&keptnevaluation.KeptnEvaluationReconciler{
+			Client:   k8sManager.GetClient(),
+			Scheme:   k8sManager.GetScheme(),
+			Recorder: k8sManager.GetEventRecorderFor("test-evaluation-controller"),
+			Log:      GinkgoLogr,
+			Meters:   initKeptnMeters(),
+			Tracer:   tracer.Tracer("test-evaluation-tracer"),
+		}}
+		setupManager(controllers) // we can register multiple time the same controller
+		ns = makeKLTDefaultNamespace(providers.KLTNamespace)
+	})
+
+	BeforeEach(func() { // list var here they will be copied for every spec
+		evaluationName = "test-evaluation"
+		evaluationDefinitionName = "my-evaldefinition"
+		metricName = "metric1"
+		namespaceName = "default" // namespaces are not deleted in the api so be careful
+	})
+
+	Describe("Testing reconcile Evaluation scenario when using KeptnMetric instead of provider directly", func() {
```

Comment:
this is not really important but it bugs me that the test reports comes out with badly formatted sentences:

KeptnEvaluationController Should succeed, as it finds valid values in KeptnMetric ✓
KeptnEvaluationController Metric status does not exist ✓
KeptnEvaluationController Metric does not exist ✓
KeptnEvaluationController Invalid provider ✓

Could you make them pretty? 👼